### PR TITLE
First implementation test sc

### PR DESCRIPTION
**Pull request by cyrannano**
Changed:

- [x] Visitor classes implementation
- [x] Controllers for endpoints handling (scenario loading, scenario steps counting)
- [x] Main function now has a @bean that creates a shared object of ScenarioQualityChecker

Give it a whirl and see if it actually works.

To test first make a POST request to /scenarioDataLoad and give it a scenario JSON structure.
Then you can make a GET request on /allStepCount to get the scenario step count.

### CODE REVIEW

File: src/main/java/pl/put/poznan/checker/logic/AllStepCountingVisitor.java [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
+    public int visit(SimpleStep simpleStep) {
+        return 1;
     }
 
     @Override
-    public int visitComplexStep(ComplexStep complexStep) {
-        return 0;
+    public int visit(ComplexStep complexStep) {
+        int stepCount = 0;
+        AllStepCountingVisitor scv = new AllStepCountingVisitor();
```
Comment (Drske) [inline src/main/java/pl/put/poznan/checker/logic/AllStepCountingVisitor.java]: Creating new object is not necessary.

File: src/main/java/pl/put/poznan/checker/logic/AllStepCountingVisitor.java [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
+        for (Step st : complexStep.subscenario) {
+            stepCount += st.accept(scv);
```
Comment (Drske) [inline src/main/java/pl/put/poznan/checker/logic/AllStepCountingVisitor.java]: `stepCount += st.accept(this)`

should be sufficient.

**Comment by Drske** [comment]
It's worth considering to transfer all logic responsibility to visitors.

In such a case it would be nice to add `visit(MainScenario mainScenario)` method to a `Visitor`. Then:

In `AllStepCountingVisitor` add:
```
    public int visit(MainScenario mainScenario) {
        int stepCount = 0;

        for (Step step : mainScenario.steps) {
            stepCount += step.accept(this);
        }
        return stepCount;
    }
```

In `MainScenario` add:

```
public int accept(ScenarioVisitor visitor) {
        return visitor.visit(this);
    }
```

In `ScenarioQualityChecker` modify:

```
    public int countAllSteps(){
        return scenario.accept(new AllStepCountingVisitor());
    }
```

**Review by Drske — APPROVED** [review]

**Comment by cyrannano** [comment]
[Check number of steps with keywords in a scenario](https://trello.com/c/meI8Ttn9/14-check-number-of-steps-with-keywords-in-a-scenario)

**Comment by cyrannano** [comment]
[Check number of steps in a scenario](https://trello.com/c/NpONmdQU/11-check-number-of-steps-in-a-scenario)

**Pull request merged**